pkg/targets/adapter/ibmmqtarget/adapter.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2021 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ibmmqtarget
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	pkgadapter "knative.dev/eventing/pkg/adapter/v2"
+	"knative.dev/pkg/logging"
+
+	targetce "github.com/triggermesh/triggermesh/pkg/targets/adapter/cloudevents"
+	"github.com/triggermesh/triggermesh/pkg/targets/adapter/ibmmqtarget/mq"
+)
+
+var _ pkgadapter.Adapter = (*ibmmqtargetAdapter)(nil)
+
+type ibmmqtargetAdapter struct {
+	replier  *targetce.Replier
+	ceClient cloudevents.Client
+	logger   *zap.SugaredLogger
+	mqEnvs   *TargetEnvAccessor
+	reply    *mq.ReplyTo
+	queue    *mq.Object
+}
+
+// NewAdapter adapter implementation
+func NewAdapter(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClient cloudevents.Client) pkgadapter.Adapter {
+	env := envAcc.(*TargetEnvAccessor)
+	logger := logging.FromContext(ctx)
+
+	replier, err := targetce.New(env.Component, logger.Named("replier"),
+		targetce.ReplierWithStatefulHeaders(env.BridgeIdentifier),
+		targetce.ReplierWithStaticResponseType("io.triggermesh.ibmmqtarget.response"),
```

Comment:
normally we put the event types in the `/pkg/apis/targets/v1alpha1/` folder under the targets lifecycle file. see here -> https://github.com/triggermesh/triggermesh/blob/f879f93c7d40c0d9f5b5bf9018d11bb545fb6192/pkg/apis/targets/v1alpha1/googlecloudfirestore_lifecycle.go#L29

Reply:
I see now. you actually do have these pre-existing.. can we use those instead?